contracts/CurrencyNetwork.sol

```diff
@@ -671,21 +650,18 @@ contract CurrencyNetwork is CurrencyNetworkInterface, Authorizable, debtTracking
         bool reducingDebtOfNextHopOnly = true;
 
         // check path in reverse to correctly accumulate the fee
-        for (uint i = _path.length; i > 0; i--) {
-            // the address of the receiver is _path[i-1]
-            address sender;
+        for (uint i = _path.length - 1; i > 0; i--) {
+            // the address of the receiver is _path[i]
```

Comment:
we can maybe use a variable here?
Pro cons? Gas usage? Not enough stack slots?

Reply:
Not enough stack slots.

Reply:
It actually works now, so we could.

Reply:
It is about 800 gas cheaper with the variables for a transfer with 3 mediators.

Reply:
And it's about 400 cheaper with the variable for a transfer with 3 mediators for transferReceiverPays

Reply:
good